layered-architectures-module-2/Carting.DAL/Repositories/Implementations/CartRepository.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Carting.DAL.Models;
+using Carting.DAL.Repositories.Interfaces;
+using MongoDB.Driver;
+
+namespace Carting.DAL.Repositories.Implementations;
+
+public class CartRepository : MongoRepositoryBase<Cart>, ICartRepository
+{
+    public CartRepository(IMongoDatabase mongoDatabase) : base(mongoDatabase)
+    {
+    }
+
+    public virtual Task<Cart> GetCartByExternalIdAsync(int id)
+    {
+        return _itemCollection.FindAsync(Builders<Cart>.Filter.Eq(c => c.ExternalId, id)).Result.SingleOrDefaultAsync();
```

Comment:
this and DeleteByExternalIdAsync method.

It might be a bit cleaner way to call `FindAsync` and `Delete` methods of the `MongoRepositoryBase` passing some predicates as parameter (as an example: `FindSingleAsync(Builders<Cart>.Filter.Eq(c => c.ExternalId, id))`)

Reply:
Fixed